doc/Hammer-Use/Hammer-Config.rst

```diff
@@ -0,0 +1,193 @@
+.. _config:
+
+The hammer-config library
```

Comment:
I think this section is better described as "Hammer IR and Meta Variables" and the hammer-config library can be a subsection explaining the Hammer IR <-> JSON/YAML translation.

Perhaps the `get_setting` referenced many times later can also be introduced as being available to everything (technologies, tool steps, hooks).